internal/ingress/controller/controller.go

```diff
@@ -231,12 +235,21 @@ type KongController struct {
 	PluginSchemaStore PluginSchemaStore
 
 	Logger logrus.FieldLogger
+
+	tmpDir string
 }
 
 // Start starts a new master process running in foreground, blocking until the next call to
 // Stop.
 func (n *KongController) Start() {
 	n.Logger.Debugf("startin up controller")
+	if len(n.cfg.DumpConfig) > 0 {
+		var err error
+		n.tmpDir, err = ioutil.TempDir("", "controller")
+		if err != nil {
+			panic(fmt.Errorf("failed to create a temporary working directory: %v", err))
+		}
+	}
```

Comment:
There are many similar initializations in the body of `main()`. I recommend moving this block there.